src/main/java/org/cardanofoundation/cfexploreraggregator/txcount/processor/AddressTxCountProcessor.java

```diff
@@ -33,52 +43,79 @@ public class AddressTxCountProcessor {
 
     private final AddressTxCountRepository addressTxCountRepository;
 
+    private final ConcurrentHashMap<String, Tuple<Long, Long>> hashCounts = new ConcurrentHashMap<>();
+
+    private final ScheduledExecutorService dbWriteScheduler = Executors.newScheduledThreadPool(1);
+
+    @Value("${explorer.aggregation.addressTxCount.Safe-Slot-Distance}")
+    private long safeSlotDistance;
+
+    @PostConstruct
+    private void init() {
+        // flushing every 10 seconds to the DB
+        dbWriteScheduler.scheduleAtFixedRate(this::flushToDatabase, 10, 10, TimeUnit.SECONDS);
```

Comment:
While this is a clever way of continuosly push data on db, Satya has designed this approach to keep in sync both event processing and cursor

https://github.com/bloxbean/yaci-store/blob/main/stores/utxo/src/main/java/com/bloxbean/cardano/yaci/store/utxo/processor/AddressProcessor.java

The issue that this approach solves is during the parallel processing of blocks. Instead of processing block by block, if order matters, you can put the blocks in a "cache", and process them in order by block number once you get the CommitEvent. This keeps aggregator and cursor in sync and should be application restart resilient.

Reply:
I'll check the commitEvent, wasn't aware of this one. Good hint!